src/cgame/cg_rocket_draw.cpp

```diff
@@ -2174,6 +2174,131 @@ class PredictedMineEfficiencyElement : public HudElement
 	std::unordered_map<int, std::string> pluralSuffix;
 };
 
+class BarbsHudElement : public HudElement
+{
+public:
+	BarbsHudElement ( const Rocket::Core::String& tag ) :
+	HudElement ( tag, ELEMENT_ALIENS ),
+	numBarbs( 0 ),
+	maxBarbs( 0 ),
+	regenerationInterval ( 0 ),
+	t0 ( 0 ),
+	offset ( 0 ) {}
+
+	void OnAttributeChange( const Rocket::Core::AttributeNameList& changed_attributes )
+	{
+		HudElement::OnAttributeChange( changed_attributes );
+		if ( changed_attributes.find( "src" ) != changed_attributes.end() )
+		{
+			maxBarbs = BG_Weapon( WP_ALEVEL3_UPG )->maxAmmo;
```

Comment:
You could also derive the weapon held from `cg` if I remember correctly, this would allow the code to work for different alien classes that get some notion of barbs to attack with in the future.